### PR TITLE
Migrate precommit rules

### PR DESCRIPTION
**Pull request by davidbarna**
## Description
Migration of packages of https://github.com/SUI-Components/frontend-pre-commit-rules/tree/master/packages/frontend-pre-commit-rules

Please review @carlosvillu and @miduga 

---

@carlosvillu in the monorepo I've only implemented sui-lint, no sui-precommit. I really think we should change the concept/API. To me if should be a CLI "sui-precommit run" that executes all the rules (lint, test, etc) and  command "sui-precommit install" that would install it. And, install should only install husky **if not installed** en set one line in package json:

`
{
   "precommit": "sui-precommit run"
}
`

Much cleaner and flexible, don't you think ?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,10 +11,13 @@
     "co": "sui-mono commit",
     "release": "sui-mono release",
     "release:check": "sui-mono check",
-    "commitmsg": "validate-commit-msg"
+    "lint": "sui-lint js",
+    "commitmsg": "validate-commit-msg",
```
Comment (carlosvillu) [inline package.json]: Can we remove this line ?!

File: packages/sui-precommit/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@schibstedspain/sui-precommit",
+  "version": "1.1.0",
+  "description": "Installs a git hook that executes sui commit rules",
+  "main": "index.js",
+  "scripts": {
+    "build": "echo \"building...\"",
+    "install": "./install.js"
```
Comment (miduga) [inline packages/sui-precommit/package.json]: That's being fixed on the actual pre-commit-rules for Windows.

`"install": "node ./install.js"`

File: package.json [hunk 3]
```diff
@@ -11,10 +11,13 @@
...
+    "precommit": "npm run lint"
   },
   "devDependencies": {
     "@schibstedspain/sui-mono": "1.1",
+    "@schibstedspain/sui-lint": "2.0",
```
Comment (carlosvillu) [inline package.json]: Change the version only to major

File: packages/sui-cz/config.js [hunk 4]
```diff
@@ -3,13 +3,13 @@ const readdirSync = require('fs').readdirSync
 const statSync = require('fs').statSync
 
 const basePath = process.cwd()
-const package = require(path.join(basePath, 'package.json'))
+const packageConfig = require(path.join(basePath, 'package.json')).config
```
Comment (carlosvillu) [inline packages/sui-cz/config.js]: Are we sure that always will be a config key in the package ?!

File: packages/sui-precommit/package.json (hunk 2 above)
Comment (miduga) [inline packages/sui-precommit/package.json]: :+1:

File: packages/sui-lint/bin/sui-lint-sass.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+const child = execFile(
+  SASS_LINT_PATH,
+  [
+    '-c', CONFIG_PATH,
+    '-i', '"node_modules, lib, dist"',
+    '-v',
+    'src/**/*.scss'
+  ].concat(args),
+  err => err && err.code && process.exit(err.code)
```
Comment (carlosvillu) [inline packages/sui-lint/bin/sui-lint-sass.js]: Maybe we have lost the lastest change she made in the repo by @julian

File: package.json (hunk 1 above)
Comment (davidbarna) [inline package.json]: Not yet.

File: package.json (hunk 3 above)
Comment (davidbarna) [inline package.json]: What do you mean ?

File: packages/sui-lint/bin/sui-lint-sass.js (hunk 5 above)
Comment (davidbarna) [inline packages/sui-lint/bin/sui-lint-sass.js]: I'll review them before deprecated, but first a want to merge this.

File: packages/sui-cz/config.js (hunk 4 above)
Comment (davidbarna) [inline packages/sui-cz/config.js]: Not, but package, yet... No it doesn't matter, right ? I'm not changing the behavior at all, which is what matters. This was change due to linting...

File: package.json (hunk 3 above)
Comment (miduga) [inline package.json]: `"@schibstedspain/sui-lint": "2"` use the major directly.

File: packages/sui-cz/config.js (hunk 4 above)
Comment (miduga) [inline packages/sui-cz/config.js]: But in the is accessing the config inside the config? Is that right? https://github.com/SUI-Components/sui/pull/3/files/2e1548d79021427651f226ae1189b2082f473b6b#diff-47efdbae34d9cf15c07739301961b08eR10

File: package.json (hunk 3 above)
Comment (davidbarna) [inline package.json]: Ok I'm still not used to comver... Right!

File: packages/sui-cz/config.js (hunk 4 above)
Comment (davidbarna) [inline packages/sui-cz/config.js]: @miduga No it's not !!!! Refactor mess alert !!!! :trollface:

File: package.json (hunk 3 above)
Comment (davidbarna) [inline package.json]: @miduga fixed.

File: packages/sui-cz/config.js (hunk 4 above)
Comment (davidbarna) [inline packages/sui-cz/config.js]: @miduga fixed.

**Pull request merged**